pkg/awsapi/iam.go

```diff
@@ -292,9 +302,47 @@ func (i *IAM) AttachInlineRolePolicy(ctx context.Context, req IAMRoleRequest) (*
 	return &IAMRoleResponse{}, nil
 }
 
+// AttachManagedRolePolicy function attaches managed policy to the role
+func (i *IAM) AttachManagedRolePolicy(ctx context.Context, policyName string, roleName string) error {
+
+	policyARN := aws.String(fmt.Sprintf("arn:aws:iam::%s:policy/%s", AwsAccountId, policyName))
+
+	_, err := i.Client.AttachRolePolicy(&iam.AttachRolePolicyInput{
+		RoleName:  aws.String(roleName),
+		PolicyArn: policyARN,
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case iam.ErrCodeEntityAlreadyExistsException:
+				fmt.Println(iam.ErrCodeEntityAlreadyExistsException, aerr.Error())
+			case iam.ErrCodeLimitExceededException:
+				fmt.Println(iam.ErrCodeLimitExceededException, aerr.Error())
+			case iam.ErrCodeNoSuchEntityException:
+				fmt.Println(iam.ErrCodeNoSuchEntityException, aerr.Error())
+			case iam.ErrCodeServiceFailureException:
+				fmt.Println(iam.ErrCodeServiceFailureException, aerr.Error())
+			default:
+				fmt.Println(aerr.Error())
+			}
+		}
+		return err
+	}
+	return nil
+}
+
 //DeleteRole function deletes the role in the account
 func (i *IAM) DeleteRole(ctx context.Context, roleName string) error {
 
+	// Detach managed policies
+	for _, policy := range ManagedPolicies {
```

Comment:
Check out a way if you can get the list of managed policies attached to the role? There might be a cases where people updating the role in the AWS IAM console directly and that leaves us in error state if we try to delete the role.
Check out https://github.com/keikoproj/iam-manager/issues/11. we can probably fix this issue as part of that

Reply:
I will do that